src/controller/gameController.ts

```diff
@@ -1,47 +1,85 @@
 import { AuthRequest } from './authenticationController.js';
 import {Response, NextFunction} from 'express';
 import { AppError, errorWrapper } from '../utils/errors.js';
+import { v4 as uuidv4 } from 'uuid';
 
 import { LocalStorage } from '../LocalStorage/localStorage.js';
+import { em } from '../database.js';
+import { Game, Player, User } from '../entity/index.js';
+import { Card } from '../LocalStorage/Card.js';
+import { InviteResponse, InviteType } from '../LocalStorage/InviteResponse.js';
+import assert from 'assert';
 import { NopeSocket } from '../router/socketRouter.js';
 
-export const gameInvite =  errorWrapper(async (req: AuthRequest, res:Response, next:NextFunction) => {
-    // 1) Convert the input to ConnectedUser Objects
+export const gameInvite = errorWrapper(async (req: AuthRequest, res:Response, next:NextFunction) => {
+    // 1) Convert the input to a Game Objects
     if (!req.body.players || !Array.isArray(req.body.players)) {
         return next(new AppError('Please provide an array of players', 400));
     }
-
     if (req.body.players.length < 2) {
         return next(new AppError('Please provide at least two players', 400));
     }
 
-    // 2) Check if the players are available
-    const convertedSocket : NopeSocket[] = convertToSocket(req.body.players);
-
+    const game = new Game(uuidv4(), Card.getRandomCard());
+    const userRepository = em().getRepository(User);
+    for (let playerIndex = 0; playerIndex < req.body.players.length; playerIndex++) {
+        const user : User | null = await userRepository.findOne({'username':req.body.players[playerIndex].username});
+        if (user === null) {
+            return next(new AppError(`User ${req.body.players[playerIndex].username} not found`, 400));
+        }
+        const player = new Player(user.username, game, playerIndex);
+        player.tempUser = user;
+        game.players.add(player);
+    }
 
-    // TODO: Create a new game object here and send it to the clients
+    // 2) Check if the players are available
+    if(!await game.populatePlayerSockets(LocalStorage.getInstance().freeSockets)) {
+        throw new AppError('Player not registered or currently in game.', 400);

```

Comment:
Machmal nehmen wir zum Error werfen `throw` und manchmal benutzen wir `return next`. In Funktionen, in denen wir keinen Zugriff auf `next` haben können wir `throw` verwenden. Aber wenn wir Zugriff auf `next` haben, sollten wir auch `next` bennutzen.